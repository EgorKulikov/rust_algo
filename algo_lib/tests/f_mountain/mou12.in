20000
I 9320 9327 241221
I 9317 9351 24169
I 9311 9358 344226
I 9260 9363 453107
I 9270 9372 369440
I 9258 9407 0
I 9237 9427 266283
I 9213 9444 116817
I 9211 9467 296808
I 9206 9480 0
I 9206 9487 15143
I 9205 9495 14364
I 9194 9499 0
I 9182 9512 80169
I 9175 9520 441060
I 9175 9521 103280
I 9166 9574 0
Q 0
I 9141 9591 186409
I 9137 9592 209456
I 9137 9592 167668
I 9128 9607 22866
I 9111 9637 427742
I 9107 9642 197803
I 9085 9660 474374
I 9084 9665 202264
I 9081 9671 108371
I 9080 9696 394723
I 9077 9714 0
I 9058 9721 385847
I 9046 9723 468314
I 9033 9726 51346
I 9017 9728 397324
I 9013 9729 0
I 8961 9762 165828
I 8960 9782 350431
I 8944 9804 411923
I 8949 9840 210726
I 8945 9855 116713
I 8944 9856 28739
I 8919 9864 277790
I 8902 9877 150743
I 8878 9879 364139
I 8886 9883 342155
I 8874 9895 461022
I 8857 9899 494308
I 8853 9901 3470
I 8837 9912 334558
I 8835 9926 235829
I 8834 9928 239880
I 8817 9936 314271
I 8815 9948 173753
I 8815 9968 33002
I 8807 9982 242012
I 8802 9997 141813
I 8798 10000 7375
I 8776 10036 214532
I 8787 10041 300184
I 8772 10042 172354
I 8771 10047 0
I 8764 10058 186030
I 8750 10060 140667
I 8761 10060 81846
I 8750 10062 353611
Q 50163513
I 8720 10072 453406
I 8704 10086 17930
I 8701 10144 204041
Q 76077611
I 8701 10160 85584
I 8696 10188 278657
I 8688 10196 91010
I 8694 10198 164324
I 8664 10203 56446
I 8653 10203 12009
I 8648 10204 298719
I 8604 10209 168857
Q 138747959
I 8603 10209 243287
I 8597 10210 13283
I 8596 10222 222328
I 8573 10233 348102
I 8558 10241 299112
I 8557 10242 232464
I 8546 10262 162372
I 8545 10268 0
I 8538 10271 35722
I 8537 10273 454384
I 8535 10280 164678
I 8521 10293 93097
I 8505 10286 439172
I 8490 10307 235118
I 8502 10312 35707
I 8460 10316 469672
I 8448 10325 191405
I 8447 10343 496631
I 8430 10368 51517
I 8423 10361 45015
I 8404 10390 450036
I 8378 10418 389704
I 8376 10468 19312
I 8369 10434 35620
I 8363 10498 168360
I 8350 10507 430579
I 8350 10513 0
I 8349 10523 0
I 8348 10533 457456
I 8347 10541 318919
I 8341 10550 452488
I 8333 10557 0
I 8340 10558 382202
I 8324 10585 206504
I 8324 10622 354233
I 8310 10652 181314
I 8306 10658 209224
I 8297 10666 16604
I 8295 10670 420875
I 8278 10678 294946
I 8270 10691 241244
I 8261 10695 409370
I 8243 10710 158299
I 8235 10731 180415
I 8230 10736 398883
Q 505604788
Q 486795581
I 8219 10753 244006
I 8217 10768 150086
I 8212 10771 106148
I 8207 10781 240636
I 8163 10790 0
I 8193 10783 385951
I 8141 10796 190672
I 8139 10802 375375
I 8135 10804 260988
I 8130 10822 0
I 8122 10824 350180
I 8114 10851 191566
I 8091 10828 16748
I 8085 10855 360880
I 8081 10858 184154
I 8061 10871 105923
I 8040 10874 99675
I 8034 10882 351000
I 8013 10886 0
I 8014 10891 347463
I 8012 10900 314774
I 8000 10905 344115
I 7999 10907 335660
I 7972 10907 340599
I 7966 10947 184387
I 7961 10955 130606
I 7949 10956 332446
I 7915 10961 93756
I 7903 10971 325839
I 7898 10982 73292
I 7897 11012 278756
I 7887 11017 319386
I 7885 11018 0
I 7883 11019 318775
I 7877 11065 93803
I 7880 11078 15493
I 7876 11086 126580
I 7873 11091 54731
I 7857 11091 197312
I 7856 11104 307787
I 7849 11140 101279
I 7841 11160 301204
Q 448773049
I 7825 11174 129134
I 7788 11177 294985
I 7788 11184 212990
I 7773 11184 83544
I 7765 11189 44207
I 7763 11202 82922
I 7760 11191 44085
I 7761 11203 290431
I 7728 11256 283366
I 7712 11263 281531
I 7695 11292 277932
I 7683 11288 45017
I 7681 11294 10492
I 7658 11301 26670
I 7654 11307 225623
I 7641 11312 272331
I 7637 11324 271149
I 7634 11357 268528
I 7625 11361 267594
I 7625 11396 219438
I 7603 11405 187952
I 7591 11396 262298
I 7583 11410 261233
I 7546 11451 52011
I 7541 11455 126301
I 7532 11466 254129
I 7525 11473 156743
I 7498 11494 250187
I 7485 11502 248880
I 7477 11496 28278
I 7469 11507 191361
I 7469 11543 158394
I 7468 11605 0
I 7468 11617 174608
I 7467 11618 240847
I 7465 11647 239062
I 7464 11685 236854
I 7438 11685 106280
I 7436 11696 183614
I 7437 11699 16887
I 7405 11730 15139
I 7431 11738 232034
I 7401 11746 111904
I 7401 11750 229885
I 7387 11753 93080
I 7386 11759 107784
I 7381 11774 227583
I 7377 11789 33762
I 7369 11808 225225
I 7371 11826 224315
I 7366 11832 23456
I 7362 11847 222915
Q 959031289
I 7349 11865 206984
I 7342 11853 218312
I 7334 11879 2787
I 7322 11883 219202
I 7314 11895 39747
Q 181411714
I 7310 11914 209530
I 7304 11924 0
I 7297 11930 215796
I 7294 11934 8065
I 7267 11939 213995
I 7260 11951 7144
I 7257 11970 212134
I 7254 11970 211999
I 7254 11972 69339
I 7242 11995 210349
I 7223 12009 0
I 7221 12012 208681
I 7197 12048 79779
I 7211 12081 205067
I 7182 12086 11015
I 7181 12126 202183
I 7169 12126 180726
I 7169 12128 201612
I 7164 12133 19012
I 7155 12138 200642
I 7153 12139 200521
I 7152 12153 199920
I 7137 12169 51334
I 7134 12161 198804
I 7087 12186 166743
I 7085 12201 2110
I 7083 12202 195312
I 7072 12207 194704
I 7070 12211 54898
I 7049 12235 192789
I 7047 12239 192566
I 7031 12251 34684
I 7029 12266 190912
I 7020 12300 93808
I 7016 12306 92749
I 7003 12320 188040
I 7002 12326 187793
I 7002 12357 157889
I 7000 12333 186766
I 6989 12377 185563
I 6983 12388 54054
I 6975 12392 184569
I 6964 12398 0
Q 0
I 6963 12418 183284
I 6952 12420 182848
I 6951 12435 182315
I 6946 12425 100278
I 6944 12437 182016
I 6933 12451 181192
I 6927 12460 180701
I 6924 12462 13185
I 6916 12462 180277
I 6910 12465 179985
I 6910 12468 114519
I 6898 12477 34376
I 6887 12478 178826
I 6886 12482 166630
I 6872 12494 124721
I 6877 12499 177730
I 6855 12519 176522
I 6842 12531 98351
I 6771 12540 173310
I 6821 12536 76468
I 6767 12545 173040
I 6762 12565 172294
I 6759 12600 171174
I 6744 12611 170415
I 6737 12613 0
I 6727 12615 147363
I 6717 12626 169204
I 6693 12654 167728
I 6692 12668 0
I 6648 12671 166002
I 6622 12674 52705
I 6620 12690 164717
I 6617 12692 164581
I 6609 12674 164364
I 6591 12694 76042
I 6593 12703 163614
I 6575 12703 163158
I 6559 12704 15588
I 6546 12754 161056
I 6512 12757 106784
I 6509 12763 98422
I 6508 12766 159769
I 6474 12768 158856
I 6466 12789 158127
Q 37782674
I 6438 12790 21507
I 6420 12810 156470
I 6430 12811 156445
I 6345 12824 63291
I 6340 12833 153988
I 6328 12838 52831
I 6317 12845 153162
I 6303 12847 152788
I 6289 12851 152369
I 6279 12855 152045
I 6278 12859 151929
I 6278 12860 151906
I 6268 12862 151630
I 6258 12864 151354
I 6254 12901 0
I 6249 12930 149655
I 6243 12945 149186
I 6241 12946 149120
I 6217 12950 84803
I 6216 12963 148192
I 6195 12975 147470
I 6215 12976 147449
Q 589794206
I 6195 13013 146649
I 6193 13032 146198
I 6178 13038 0
I 6174 13050 26110
I 6149 13056 144759
I 6142 13069 144341
I 6116 13085 143472
I 6102 13106 35675
I 6093 13113 142429
I 6065 13124 141643
I 6060 13154 140944
I 6057 13156 140845
I 6050 13157 56298
I 6045 13176 140213
I 6033 13181 0
I 6027 13184 16314
I 6026 13190 31722
I 6021 13193 139411
I 6021 13195 139372
I 6019 13203 16088
I 6020 13211 139041
I 6015 13215 138869
I 6009 13237 138331
I 6001 13247 137988
I 5995 13254 0
I 5985 13258 137475
I 5967 13281 97632
I 5948 13298 117269
I 5903 13352 134228
I 5923 13365 133993
I 5870 13393 98370
I 5831 13394 132205
I 5830 13402 132048
I 5798 13406 131423
I 5790 13409 121169
I 5779 13428 0
I 5776 13438 130497
I 5763 13438 42670
Q 260500945
I 5761 13446 130106
I 5736 13454 129550
I 5688 13468 35677
I 5668 13475 128073
I 5657 13477 127860
I 5654 13515 127194
I 5651 13487 127145
I 5651 13517 127113
I 5631 13538 126454
I 5626 13538 27608
I 5608 13551 96338
I 5607 13556 125786
I 5606 13558 93697
I 5587 13586 125000
I 5575 13626 124192
I 5569 13634 123977
I 5531 13638 51421
I 5527 13644 123183
I 5520 13649 123001
I 5520 13648 123001
I 5512 13664 122654
I 5496 13676 122234
I 5465 13691 121550
I 5461 13678 121491
I 5459 13705 121256
I 5451 13707 44687
I 5421 13710 62222
Q 468222289
I 5412 13752 119889
I 5393 13779 119232
I 5388 13797 118906
I 5384 13786 118849
I 5381 13804 118708
I 5380 13818 77471
I 5359 13856 117674
I 5358 13863 117564
I 5355 13863 117522
I 5349 13885 117137
I 5327 13869 0
I 5326 13895 88700
I 5297 13896 2704
I 5262 13939 63271
Q 420746068
I 5261 13907 0
I 5234 13964 0
I 5243 13968 114600
I 5208 13981 113973
I 5198 13983 113817
I 5187 13994 113533
I 5174 14003 113250
I 5166 14017 6865
I 5165 14020 112917
I 5148 14032 112549
I 5145 14035 112473
I 5106 14058 111694
I 5104 14077 111433
I 5068 14088 0
I 5067 14091 110803
I 5066 14116 110485
I 5059 14110 94272
I 5029 14122 0
I 5012 14123 109745
I 5011 14143 109493
I 5002 14146 109349
I 4994 14152 0
I 4990 14156 109086
I 4980 14157 108956
I 4982 14178 -23
I 4970 14181 108554
I 4948 14183 108271
I 4946 14184 17121
I 4941 14185 0
I 4919 14188 107874
I 4903 14189 107677
I 4893 14198 0
I 4883 14219 107100
I 4866 14226 106826
I 4873 14260 98090
I 4863 14269 3103
I 4851 14298 105842
I 4848 14284 105808
I 4850 14302 59968
I 4844 14336 0
I 4829 14337 105163
Q 895303896
I 4818 14342 104986
I 4816 14345 104931
I 4795 14348 526
I 4791 14368 104405
Q 988015660
I 4784 14381 104188
I 4781 14413 103809
I 4768 14426 103530
I 4767 14427 103508
I 4754 14430 103337
I 4745 14435 103188
I 4735 14469 102722
I 4724 14469 102606
I 4712 14474 102427
I 4701 14485 102197
I 4687 14485 102051
I 4676 14495 101832
I 4667 14495 101739
I 4663 14502 101626
I 4659 14511 101491
I 4649 14527 101224
I 4641 14551 100897
I 4631 14560 68365
I 4627 14570 100563
I 4604 14576 100270
I 4598 14584 100130
I 4593 14589 100030
I 4592 14594 99970
I 4591 14596 6868
I 4579 14605 99730
I 4566 14628 99373
I 4540 14630 0
Q 0
I 4525 14636 0
I 4519 14643 98765
I 4516 14652 98648
I 4511 14659 98531
I 4504 14678 95143
I 4494 14684 98125
I 4493 14690 98058
I 4462 14710 97570
Q 733063815
I 4452 14746 97134
I 4438 14752 96946
I 4434 14759 0
I 4429 14763 96758
I 4423 14763 91278
I 4415 14765 56547
I 4392 14767 96376
I 4392 14772 96329
I 4391 14810 95969
I 4385 14820 95822
I 4320 14843 95020
I 4291 14843 94759
I 4257 14849 94401
I 4247 14854 38305
I 4237 14867 94064
I 4237 14883 93923
I 4233 14891 25017
I 4228 14893 93755
I 4204 14898 93501
I 4203 14905 81885
Q 37393318
I 4186 14916 93187
I 4184 14919 93144
I 4180 14923 93075
I 4161 14924 92902
I 4153 14937 92721
I 4137 14941 92549
I 4134 14953 92421
I 4130 14989 92081
I 4123 14994 91979
I 4117 15007 0
I 4104 15009 8166
I 4096 15021 91524
I 4084 15030 91349
I 4079 15032 91290
I 4073 15047 91116
I 4052 15051 90909
I 4036 15078 90555
I 4036 15093 90432
I 4024 15103 90252
I 4022 15109 90187
I 4013 15115 66040
I 4003 15128 89879
I 3999 15141 89742
I 3990 15147 19571
I 3971 15150 89445
I 3969 15159 89357
I 3961 15163 89261
I 3949 15165 89150
I 3896 15170 11629
I 3896 15171 88683
Q 860701495
I 3877 15180 88464
I 3875 15182 88432
I 3858 15198 88175
I 3861 15200 88160
I 3857 15201 0
I 3841 15223 87850
I 3814 15237 87535
I 3811 15263 87313
I 3805 15267 87237
I 3803 15288 84576
I 3803 15270 87066
I 3793 15300 86896
I 3787 15307 86798
I 3784 15313 86730
I 3777 15316 0
I 3765 15341 86378
I 3761 15343 86333
I 3745 15352 86147
I 3725 15359 40731
I 3721 15360 34212
I 3716 15362 31968
I 3708 15381 85660
Q 425853708
I 3696 15386 85535
I 3692 15399 85411
I 3692 15420 85258
Q 515998389
I 3687 15437 0
Q 0
I 3669 15443 84925
I 3662 15451 84817
I 3664 15455 84788
I 3661 15457 84767
I 3659 15470 84659
I 3633 15481 0
I 3651 15484 84502
I 3627 15487 84309
I 3610 15505 84061
I 3610 15508 84040
I 3600 15511 83948
I 3598 15527 83822
I 3590 15530 83745
I 3589 15534 81995
I 3586 15599 83236
I 3584 15605 83180
I 3574 15612 54990
I 3572 15651 6249
I 3555 15656 82630
I 3513 15676 82209
I 3511 15681 0
I 3495 15681 0
I 3486 15683 81980
I 3484 15694 69360
Q 836397167
I 3479 15729 81625
I 3457 15759 75959
I 3456 15774 0
I 3454 15782 81109
I 3445 15803 80912
I 3445 15805 0
Q 0
I 3429 15823 80677
Q 919716377
I 3412 15830 80521
I 3417 15838 80469
I 3410 15843 80424
I 3403 15843 80379
I 3394 15844 80314
I 3391 15849 80263
I 3371 15854 80102
I 3363 15864 0
I 3361 15881 79865
I 3357 15888 79795
I 3355 15889 79776
I 3337 15890 79655
I 3344 15900 79592
I 3336 15905 79554
I 3336 15906 79548
I 3323 15909 79447
I 3322 15955 79151
I 3300 15954 0
I 3299 15976 78876
I 3298 15988 78795
I 3295 15996 3337
I 3291 16005 78647
I 3287 16006 78616
I 3278 16014 78511
I 3266 16015 3673
I 3261 16025 78339
I 3252 16028 78265
I 3239 16031 78167
I 3202 16032 31592
Q 367705168
I 3201 16034 77918
I 3178 16039 0
I 3186 16041 77784
I 3170 16045 77663
I 3165 16047 77621
I 3157 16068 77447
I 3153 16084 77327
I 3136 16107 77089
I 3083 16132 76628
I 3113 16140 76581
I 3063 16211 76051
I 3059 16207 76028
I 3055 16219 75958
I 3011 16226 75665
I 2995 16250 75437
I 2991 16251 0
I 2977 16280 0
I 2973 16294 75063
I 2971 16322 74895
I 2962 16330 0
I 2957 16339 74721
I 2952 16341 0
I 2953 16363 74565
I 2946 16361 74526
I 2943 16373 74454
I 2901 16378 74194
I 2918 16401 74068
I 2896 16379 74041
I 2887 16394 73991
I 2864 16406 73838
Q 814205991
I 2852 16422 73686
I 2849 16440 73572
I 2846 16457 73464
I 2846 16460 73448
I 2833 16468 73335
I 2814 16481 73163
I 2812 16489 73110
I 2809 16492 73078
I 2777 16520 72759
I 2768 16525 72684
I 2760 16531 72611
I 2757 16531 72595
I 2743 16555 72395
I 2739 16550 72374
I 2734 16560 17557
I 2718 16571 72181
I 2715 16577 72134
I 2703 16580 72056
I 2695 16582 70424
I 2683 16607 71813
I 2681 16597 38202
I 2669 16651 0
I 2663 16673 71372
I 2638 16672 71245
I 2636 16726 70967
I 2628 16749 11025
Q 133094041
I 2623 16748 70790
Q 916143893
I 2622 16764 70706
I 2617 16768 70661
Q 214078006
I 2615 16779 70596
I 2615 16788 70551
I 2596 16791 70442
Q 464944332
I 2579 16809 70269
I 2566 16814 70180
I 2577 16819 70155
I 2556 16826 70072
I 2550 16836 69993
I 2537 16856 69832
I 2535 16866 69773
I 2532 16885 69666
Q 612542862
I 2526 16924 0
I 2525 16925 69439
I 2525 16936 69386
I 2491 16938 69213
I 2499 16944 69184
I 2493 16964 10208
I 2488 16991 68946
I 2484 16999 68889
I 2476 17010 68799
I 2442 17023 0
I 2429 17032 0
I 2426 17037 68436
I 2424 17041 0
I 2424 17046 68385
I 2422 17057 68324
I 2419 17076 68222
I 2418 17077 68212
Q 18889819
I 2410 17093 68101
I 2407 17098 65559
I 2403 17133 67884
I 2403 17136 67870
I 2375 17144 0
I 2342 17171 67430
I 2351 17177 67403
I 2319 17177 67299
I 2300 17178 21080
I 2282 17192 67064
I 2278 17203 66997
I 2278 17211 0
I 2266 17263 66675
I 2261 17263 66653
I 2222 17286 26053
I 2173 17303 66089
I 2173 17293 0
I 2173 17307 65895
I 2172 17314 19863
I 2157 17319 25409
I 2148 17322 0
I 2120 17339 65703
I 2109 17340 65651
I 2103 17343 65612
I 2101 17365 65509
I 2101 17375 65466
I 2099 17389 65397
I 2092 17391 65359
I 2092 17393 65350
I 2092 17403 65308
I 2082 17415 65214
I 2078 17431 65129
I 2076 17496 64846
I 2072 17499 64817
I 2065 17508 64750
I 2063 17509 64737
I 2045 17537 64545
Q 931567920
I 2037 17567 64387
I 2026 17568 64337
I 2019 17583 7550
I 2016 17603 64151
I 2006 17610 64082
I 2005 17610 64077
I 1997 17612 64036
I 1994 17622 63983
I 1976 17664 63738
I 1970 17668 63698
I 1953 17678 63588
I 1941 17679 63536
I 1936 17695 63451
I 1933 17749 0
I 1926 17749 0
I 1925 17776 63083
I 1917 17784 63019
I 1912 17792 62968
I 1909 17794 62948
Q 951840242
I 1908 17815 62861
I 1898 17821 52102
I 1903 17825 -16
I 1893 17830 0
I 1884 17842 0
I 1881 17858 62586
I 1870 17876 62472
I 1869 17891 62410
I 1848 17908 62262
Q 467662869
I 1845 17918 62212
I 1830 17920 62146
I 1818 17944 35893
I 1783 17951 61846
I 1775 17963 61770
I 1769 17969 61724
I 1765 17972 61697
I 1730 17986 61511
I 1726 17990 61481
I 1706 17995 61387
I 1704 17998 0
Q 0
I 1697 18016 61274
I 1659 18025 61098
I 1655 18034 61050
I 1653 18055 60964
I 1639 18065 60875
I 1632 18095 60738
I 1625 18096 60709
Q 657380033
I 1619 18143 0
I 1606 18113 60576
I 1594 18163 60350
I 1592 18184 60266
I 1568 18185 60175
I 1559 18195 0
I 1550 18210 60020
I 1546 18210 60006
I 1541 18235 0
I 1526 18244 59812
I 1493 18267 59612
I 1481 18273 0
I 1475 18290 59467
I 1472 18292 59449
I 1469 18300 36249
I 1461 18310 59347
I 1446 18311 59290
I 1431 18323 59196
I 1407 18324 59108
I 1399 18333 59049
I 1392 18333 3546
I 1390 18337 59004
I 1386 18347 58955
I 1386 18349 7789
I 1368 18352 58875
I 1364 18359 58837
I 1357 18362 58802
I 1347 18394 58657
I 1329 18399 0
I 1311 18406 58493
I 1306 18408 58469
I 1295 18417 35410
I 1253 18447 54912
I 1237 18464 58045
I 1232 18473 57997
I 1230 18492 57927
I 1217 18511 57820
I 1212 18519 57776
I 1196 18549 57623
I 1187 18553 57580
I 1175 18553 57540
I 1160 18554 57487
I 1158 18575 0
I 1150 18561 57431
Q 37522125
I 1122 18582 57270
I 1120 18582 57263
I 1107 18593 57185
I 1106 18603 57149
I 1103 18610 57116
I 1096 18642 56989
I 1090 18688 56821
I 1080 18697 56760
I 1085 18704 23508
I 1072 18710 56692
I 1046 18776 56398
Q 235919744
I 1022 18726 22809
I 1013 18787 25692
I 982 18790 56151
I 973 18820 56028
I 972 18835 55978
I 963 18837 55944
I 951 18857 55844
I 931 18873 55732
I 931 18881 55707
I 926 18886 55676
I 905 18888 55604
I 905 18900 55567
I 903 18904 55549
I 897 18940 55420
I 897 18964 20840
I 887 18965 55312
I 816 18985 55035
I 833 19004 54978
I 832 19068 54785
I 787 19077 54671
I 804 19082 54656
Q 154231269
I 770 19091 17291
I 760 19077 54577
I 756 19105 54495
I 751 19107 54475
I 750 19107 54472
I 746 19109 33420
I 740 19113 54424
I 735 19127 54368
I 733 19131 54350
I 716 19132 54297
I 710 19134 0
I 695 19137 14127
I 694 19167 54130
I 679 19169 54080
I 683 19179 54051
I 676 19222 53917
I 668 19225 53885
I 649 19266 53711
I 637 19285 53622
I 637 19270 53622
I 617 19291 53547
I 598 19303 53458
I 579 19306 53395
I 578 19325 53339
I 553 19314 53267
I 542 19349 53168
I 534 19351 53140
I 527 19375 53053
I 512 19388 8702
I 505 19390 52949
I 505 19399 52924
Q 982821403
I 495 19404 52882
I 455 19404 52770
I 444 19407 52731
I 434 19423 0
I 436 19424 0
I 430 19495 52449
I 427 19498 21616
I 412 19515 52345
I 400 19516 52309
I 392 19518 52282
I 382 19519 52252
I 376 19522 52227
I 369 19535 52173
I 366 19542 52145
I 358 19547 52110
I 347 19547 52080
I 333 19550 52034
I 332 19587 51931
I 324 19571 51910
Q 822687243
I 321 19593 51886
I 319 19624 51797
I 308 19637 51733
I 303 19641 51708
I 284 19695 51514
I 279 19710 51461
I 275 19712 0
I 253 19716 51376
I 244 19732 51310
I 244 19762 23740
I 238 19753 0
I 232 19765 51192
Q 359625578
I 227 19771 51163
I 215 19771 51132
I 212 19785 51088
I 211 19806 51030
I 205 19815 50991
I 168 19834 50846
I 163 19834 0
I 155 19909 50620
I 112 19917 50489
I 127 19950 50405
I 107 19953 50385
I 84 19964 50299
I 75 19964 50276
Q 788340393
I 61 19970 16694
I 67 19975 50223
I 56 19983 50180
I 23 19984 50095
I 22 19993 40128
I 10 19995 50035
I 2 20000 50002
Q 720845784
E
