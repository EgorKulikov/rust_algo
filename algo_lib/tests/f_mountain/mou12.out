20000
8890
9072
9424
9496
9449
9329
11663
11877
20000
6703
10193
11867
12945
11910
13341
14253
20000
11974
4658
13600
8678
9743
20000
15541
20000
14827
14840
13889
14699
15563
5645
9195
11323
2693
16476
17029
9358
20000
12452
1802
5228
3607
19074
16171
7256
15754
14417
